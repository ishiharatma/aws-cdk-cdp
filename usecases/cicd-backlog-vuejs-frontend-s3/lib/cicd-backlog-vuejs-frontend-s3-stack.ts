import * as cdk from 'aws-cdk-lib';
import { StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as path from 'path';
import {
  aws_s3 as s3,
  aws_lambda as lambda,
  aws_apigateway as apigw,
  aws_iam as iam,
  aws_codedeploy as codedeploy,
  aws_codepipeline as codepipeline,
  aws_codepipeline_actions as codepipeline_actions,
  aws_codebuild as codebuild,
 } from 'aws-cdk-lib';

interface CicdBacklogVuejsFrontendS3StackProps extends StackProps {
  readonly pjName: string;
  readonly envName: string;
  readonly repositoryName: string;
  readonly branchName: string;
  readonly allowedIps?: string[];
  readonly websiteBucketName?: string;
  readonly distributionId?: string;
  readonly isAutoDeleteObject?: boolean;
}


export class CicdBacklogVuejsFrontendS3Stack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: CicdBacklogVuejsFrontendS3StackProps) {
    super(scope, id, props);

    const accountId = cdk.Stack.of(this).account;

    // Create Source S3 bucket
    const bucketName = `${props.pjName}-${props.envName}-source-s3-${accountId}`;
    const bucket = new s3.Bucket(this, bucketName, {
      bucketName: bucketName,
      removalPolicy: props.isAutoDeleteObject ? cdk.RemovalPolicy.DESTROY : cdk.RemovalPolicy.RETAIN,
      autoDeleteObjects: props.isAutoDeleteObject,
    });

    // Lambda function name
    const gitCloneFunctionName = `${props.pjName}-${props.envName}-git-clone-lambda-${accountId}`;

    // create lambda Iam Role and policy
    const lambdaRole = new iam.Role(this, `${gitCloneFunctionName}-lambda-role`, {
      roleName: `${gitCloneFunctionName}-role`,
      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
    });
    lambdaRole.attachInlinePolicy(new iam.Policy(this, `${gitCloneFunctionName}-lambda-policy`, {
      policyName: `${gitCloneFunctionName}-policy`,
      statements: [
        new iam.PolicyStatement({
          effect: iam.Effect.ALLOW,
          actions: [
            'logs:CreateLogGroup',
            'logs:CreateLogStream',
            'logs:PutLogEvents',
          ],
          resources: [
            `arn:aws:logs:${cdk.Stack.of(this).region}:${accountId}:log-group:/aws/lambda/${props.pjName}-${props.envName}-*`,
          ],
        }),
      ],
    }));
    // attach policy to lambda role for s3
    lambdaRole.attachInlinePolicy(new iam.Policy(this, `${gitCloneFunctionName}-s3-policy`, {
      policyName: `${gitCloneFunctionName}-s3-policy`,
      statements: [
        new iam.PolicyStatement({
          effect: iam.Effect.ALLOW,
          actions: [
            's3:GetObject',
            's3:PutObject',
            's3:ListBucket',
          ],
          resources: [
            bucket.bucketArn,
            `${bucket.bucketArn}/*`,
          ],
        }),
      ],
    }));  

    // Create Lambda function for git clone
    const pipelineName = `${props.pjName}-${props.envName}-pipeline`;
    const gitCloneFunctionPath = path.join(__dirname, '../src/lambda/backlog-git-clone/python');
    const gitCloneFunction = new lambda.Function(this, gitCloneFunctionName, {
      functionName: gitCloneFunctionName,
      description: 'git clone lambda function for s3',
      runtime: lambda.Runtime.PYTHON_3_13,
      handler: 'index.lambda_handler',
      timeout: cdk.Duration.seconds(300),
      memorySize: 1024,
      role: lambdaRole,
      architecture: lambda.Architecture.ARM_64,
      code: lambda.Code.fromAsset(gitCloneFunctionPath),
      layers: [
        lambda.LayerVersion.fromLayerVersionArn(this, 'LambdaLayer', `arn:aws:lambda:ap-northeast-1:${accountId}:layer:dulwich-layer:2`),
      ],
      environment: {
        BUCKET_NAME: bucket.bucketName,
        REPOSITORY: props.repositoryName,
        BRANCH: props.branchName,
        ZIP_FILE_NAME: "source",
        USER: "",
        PASS: "",
        PIPELINE_NAME: pipelineName,
      },
    });

    // Create API Gateway for Lambda
    const api = new apigw.LambdaRestApi(this, 'GitCloneApi', {
      restApiName: 'GitCloneApi',
      description: 'This service is for git clone',
      handler: gitCloneFunction,
      deployOptions: {
        stageName: 'dev',
      },
      policy: props.allowedIps ? new iam.PolicyDocument({
        statements: [
          new iam.PolicyStatement({
            effect: iam.Effect.DENY,
            principals: [new iam.AnyPrincipal()],
            actions: ['execute-api:Invoke'],
            resources: ['execute-api:/*/*/*'], // ステージ作成前なのでワイルドカードを使用
            conditions: {
              'NotIpAddress': {
                'aws:SourceIp': props.allowedIps,
              },
            },
          }),
          new iam.PolicyStatement({
            effect: iam.Effect.ALLOW,
            principals: [new iam.AnyPrincipal()],
            actions: ['execute-api:Invoke'],
            resources: ['execute-api:/*/*/*'],
          }),
        ],
      }) : undefined,
    });
    // add resource /backlog-webhook
    // https://[API_ID].execute-api.[REGION].amazonaws.com/dev/backlog-git-webhook
    const webhook = api.root.addResource('backlog-git-webhook');
    // add method POST
    webhook.addMethod('POST');

    // output api url
    new cdk.CfnOutput(this, 'ApiUrl', { value: api.url });

    // 静的ウェブサイトのバケット
    let websiteBucket: s3.IBucket;
    if (props.websiteBucketName) {
      // 既存のウェブサイトバケットを参照
      websiteBucket = s3.Bucket.fromBucketName(this, 'ExistingWebsiteBucket', props.websiteBucketName);
    } else {
      // 新しいウェブサイトバケットを作成
      const websiteBucketName = `${props.pjName}-${props.envName}-website-${accountId}`;
      websiteBucket = new s3.Bucket(this, websiteBucketName, {
        bucketName: websiteBucketName,
        websiteIndexDocument: 'index.html',
        websiteErrorDocument: 'index.html',
        publicReadAccess: true,
        removalPolicy: props.isAutoDeleteObject ? cdk.RemovalPolicy.DESTROY : cdk.RemovalPolicy.RETAIN,
        autoDeleteObjects: props.isAutoDeleteObject,
        cors: [
          {
            allowedMethods: [
              s3.HttpMethods.GET,
              s3.HttpMethods.HEAD,
            ],
            allowedOrigins: ['*'],
            allowedHeaders: ['*'],
          },
        ],
      });
    }

    // パイプラインの変数定義を作成
    const repositoryNameVar = new codepipeline.Variable({
      variableName: 'REPOSITORY_NAME',
      description: 'The name of the repository',
      defaultValue: props.repositoryName || '',
    });
    const branchNameVar = new codepipeline.Variable({
      variableName: 'BRANCH_NAME',
      description: 'The branch name of the repository',
      defaultValue: props.branchName || '',
    });
    const commitIdVar = new codepipeline.Variable({
      variableName: 'COMMIT_ID',
      description: 'The commit ID of the repository',
      defaultValue:'',
    });

    // Create CodePipeline
    const pipeline = new codepipeline.Pipeline(this, 'Pipeline', {
      pipelineName: pipelineName,
      pipelineType: codepipeline.PipelineType.V2,
      restartExecutionOnUpdate: true,
      variables: [repositoryNameVar, branchNameVar, commitIdVar],
    });

    // Source stage
    const sourceOutput = new codepipeline.Artifact('SourceOutput');
    const sourceAction = new codepipeline_actions.S3SourceAction({
      actionName: 'S3Source',
      bucket: bucket,
      bucketKey: 'source.zip',
      output: sourceOutput,
      trigger: codepipeline_actions.S3Trigger.NONE, // Webhook Lambda から手動でトリガー
    });
    // Add stages to pipeline
    pipeline.addStage({
      stageName: 'Source',
      actions: [sourceAction],
    });

    // Create CodeBuild project
    const buildProject = new codebuild.PipelineProject(this, 'BuildProject', {
      projectName: `${props.pjName}-${props.envName}-build`,
      environment: {
        buildImage: codebuild.LinuxBuildImage.AMAZON_LINUX_2_4,
        privileged: true,
      },
      buildSpec: codebuild.BuildSpec.fromSourceFilename('buildspec.yml'),
      cache: codebuild.Cache.local(codebuild.LocalCacheMode.CUSTOM),
//      buildSpec: codebuild.BuildSpec.fromObject({
//        version: '0.2',
//        phases: {
//          install: {
//            'runtime-versions': {
//              nodejs: '18',
//            },
//            commands: [
//              'echo Installing dependencies...',
//              'npm install',
//            ],
//          },
//          build: {
//            commands: [
//              'echo Building Vue.js application...',
//              'npm run build',
//            ],
//          },
//          post_build: {
//            commands: [
//              'echo Build completed successfully',
//            ],
//          },
//        },
//        artifacts: {
//          'base-directory': 'dist',
//          files: [
//            '**/*',
//          ],
//        },
//      }),
    });

    // Build stage
    const buildOutput = new codepipeline.Artifact('BuildOutput');
    const buildAction = new codepipeline_actions.CodeBuildAction({
      actionName: 'Build',
      project: buildProject,
      input: sourceOutput,
      outputs: [buildOutput],
    });
    pipeline.addStage({
      stageName: 'Build',
      actions: [buildAction],
    });

    // Deploy stage
    const deployAction = new codepipeline_actions.S3DeployAction({
      actionName: 'PublishAssets',
      bucket: websiteBucket,
      input: buildOutput,
      accessControl: s3.BucketAccessControl.PUBLIC_READ,
      extract: true,
    });

    pipeline.addStage({
      stageName: 'Deploy',
      actions: [deployAction],
    });

    // Add permissions for pipeline to access S3 bucket
    bucket.grantRead(pipeline.role);
    websiteBucket.grantWrite(pipeline.role);

    // Cleanup S3 buclet
    const cleanupFunctionName = `${props.pjName}-${props.envName}-cleanup-lambda-${accountId}`;
    const lambdaRoleCleanup = new iam.Role(this, `${cleanupFunctionName}-lambda-role`, {
      roleName: `${cleanupFunctionName}-role`,
      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
    });    
    lambdaRoleCleanup.attachInlinePolicy(new iam.Policy(this, `${cleanupFunctionName}-lambda-policy`, {
      policyName: `${cleanupFunctionName}-policy`,
      statements: [
        new iam.PolicyStatement({
          effect: iam.Effect.ALLOW,
          actions: [
            'logs:CreateLogGroup',
            'logs:CreateLogStream',
            'logs:PutLogEvents',
          ],
          resources: [
            `arn:aws:logs:${cdk.Stack.of(this).region}:${accountId}:log-group:/aws/lambda/${props.pjName}-${props.envName}-*`,
          ],
        }),
      ],
    })); 
    lambdaRoleCleanup.attachInlinePolicy(new iam.Policy(this, `${cleanupFunctionName}-s3-policy`, {
      policyName: `${cleanupFunctionName}-s3-policy`,
      statements: [
        new iam.PolicyStatement({
          effect: iam.Effect.ALLOW,
          actions: [
            's3:DeleteObject',
            's3:DeleteBucket',
          ],
          resources: [
            bucket.bucketArn,
            `${bucket.bucketArn}/*`,
          ],
        }),
      ],
    }));
    const cleanupFunctionPath = path.join(__dirname, '../src/lambda/cleanup/python');
    const cleanupFunction = new lambda.Function(this, cleanupFunctionName, {
      functionName: cleanupFunctionName,
      description: 'cleanup lambda function for s3',
      runtime: lambda.Runtime.PYTHON_3_13,
      handler: 'index.lambda_handler',
      timeout: cdk.Duration.minutes(15),
      memorySize: 1024,
      role: lambdaRoleCleanup,
      architecture: lambda.Architecture.ARM_64,
      code: lambda.Code.fromAsset(cleanupFunctionPath),
      environment: {
        BUCKET_NAME: bucket.bucketName,
      },
    });
    pipeline.addStage({
      stageName: 'Cleanup',
      actions: [
        new codepipeline_actions.LambdaInvokeAction({
          actionName: 'Cleanup',
          lambda: cleanupFunction,
          userParameters: {
            BUCKET_NAME: bucket.bucketName,
          },
        }),
      ],
    });

    // CloudFront Cache Invalidation
    if (!props.distributionId) {
      const cloudFrontInvalidationFunctionName = `${props.pjName}-${props.envName}-cloudfront-invalidation-lambda-${accountId}`;
      const cloudFrontInvalidationFunction = new lambda.Function(this, cloudFrontInvalidationFunctionName, {
        functionName: cloudFrontInvalidationFunctionName,
        description: 'CloudFront Cache Invalidation Lambda',
        runtime: lambda.Runtime.PYTHON_3_13,
        handler: 'index.lambda_handler',
        timeout: cdk.Duration.seconds(300),
        memorySize: 1024,
        role: lambdaRole,
        code: lambda.Code.fromAsset(path.join(__dirname, '../src/lambda/cloudfront-invalidation/python')),
        environment: {
          DISTRIBUTION_ID: '',
        },
      });
      cloudFrontInvalidationFunction.addPermission('CloudFrontInvalidationPermission', {
        principal: new iam.ServicePrincipal('lambda.amazonaws.com'),
        sourceArn: websiteBucket.bucketArn,
      });
      pipeline.addStage({
        stageName: 'CloudFrontInvalidation',
        actions: [
          new codepipeline_actions.LambdaInvokeAction({
            actionName: 'CloudFrontInvalidation',
            lambda: cloudFrontInvalidationFunction,
            userParameters: {
              DISTRIBUTION_ID: websiteBucket.bucketArn,
            },
          }),
        ],
      });
      cloudFrontInvalidationFunction.addEnvironment('DISTRIBUTION_ID', props.distributionId!);
    }
    // Add permissions for lambda to trigger pipeline
    const pipelineExecutionPolicy = new iam.Policy(this, 'PipelineExecutionPolicy', {
      statements: [
        new iam.PolicyStatement({
          effect: iam.Effect.ALLOW,
          actions: [
            'codepipeline:StartPipelineExecution',
          ],
          resources: [
            pipeline.pipelineArn,
          ],
        }),
      ],
    });
    lambdaRole.attachInlinePolicy(pipelineExecutionPolicy);

    // Update Lambda environment variables to include pipeline information
    gitCloneFunction.addEnvironment('PIPELINE_NAME', pipeline.pipelineName);
    gitCloneFunction.addEnvironment('ZIP_FILE_NAME', 'source.zip');

  }
}
