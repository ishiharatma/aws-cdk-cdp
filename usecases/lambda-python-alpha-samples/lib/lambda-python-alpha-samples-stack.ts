import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { PythonFunction, PythonFunctionProps } from "@aws-cdk/aws-lambda-python-alpha";
import {
  aws_lambda as lambda,
  aws_logs as logs  
} from "aws-cdk-lib";

export class LambdaPythonAlphaSamplesStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // Lambda関数の作成
    const lambdaProps: PythonFunctionProps = {
      runtime: cdk.aws_lambda.Runtime.PYTHON_3_13,
      index: 'index.py',
      handler: 'lambda_handler',
      timeout: cdk.Duration.seconds(30),
      memorySize: 128,
      entry: '../common/src/lambda/logger-any-loglevel/python',
      tracing: cdk.aws_lambda.Tracing.ACTIVE,
      loggingFormat: lambda.LoggingFormat.JSON,
      architecture: lambda.Architecture.ARM_64, // アーキテクチャを明示的に設定
      bundling: {
        image: lambda.Runtime.PYTHON_3_13.bundlingImage,
        platform: "linux/amd64",
        bundlingFileAccess: cdk.BundlingFileAccess.VOLUME_COPY,
        outputPathSuffix: '.',  // 出力先を明示的に指定
      },
    };
    const loglevels = [
      lambda.ApplicationLogLevel.DEBUG,
      lambda.ApplicationLogLevel.INFO,
      lambda.ApplicationLogLevel.WARN,
      lambda.ApplicationLogLevel.ERROR,
      lambda.ApplicationLogLevel.FATAL,
    ];

    loglevels.forEach((loglevel) => {
      // カスタムロググループを先に作成
      const functionName = `LambdaPythonAlphaSamples${loglevel}`;
      const logGroup = new logs.LogGroup(this, `${functionName}LogGroup`, {
        logGroupName: `/aws/lambda/${functionName}`,
        retention: logs.RetentionDays.ONE_DAY,
        removalPolicy: cdk.RemovalPolicy.DESTROY,
      });

      // Lambda関数の作成
      const lambdaFunction = new PythonFunction(this, functionName, {
        ...lambdaProps,
        functionName: functionName,
        description: `Lambda Python Alpha Samples ${loglevel}`,
        environment: {
          // 環境変数の設定
          MY_ENV_VAR: 'my_value',
        },
        applicationLogLevelV2:
          lambda.ApplicationLogLevel[
            loglevel as keyof typeof lambda.ApplicationLogLevel
          ],
        logGroup: logGroup, // 先に作成したロググループを使用
      });
      lambdaFunction.addEnvironment('MY_ENV_VAR2', 'my_value2');

      // 明示的な依存関係を追加（テストが期待する依存関係）
      const cfnLambda = lambdaFunction.node.defaultChild as cdk.aws_lambda.CfnFunction;
      const cfnLogGroup = logGroup.node.defaultChild as cdk.aws_logs.CfnLogGroup;
      cfnLambda.addDependency(cfnLogGroup);

    });
  }
}
