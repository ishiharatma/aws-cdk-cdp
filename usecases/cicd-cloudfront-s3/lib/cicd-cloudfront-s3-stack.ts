import * as cdk from 'aws-cdk-lib';
import { StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { WebPipelineConstruct } from '../../common/lib/construct-pipeline-s3web'

import * as interfaceDef from '../../common/interface/index';

interface CicdCloudFrontS3StackProps extends StackProps {
  input: interfaceDef.WebPipelineConstructProps;
}

export class CicdCloudfrontS3Stack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: CicdCloudFrontS3StackProps) {
    super(scope, id, props);

    new WebPipelineConstruct(this, 'Default', props.input);

  }
}
