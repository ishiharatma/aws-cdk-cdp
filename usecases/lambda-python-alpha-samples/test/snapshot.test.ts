// snapshot test for lambda-python-alpha-samples
import { App, Stack } from 'aws-cdk-lib';
import { Match, Template } from 'aws-cdk-lib/assertions';
import { LambdaPythonAlphaSamplesStack } from '../lib/lambda-python-alpha-samples-stack';

const projectName = 'unittest';
const envName = 'test';

const defaultEnv = {
    account: '123456789012',
    region: 'ap-northeast-1',
};

test('Snapshot', () => {
    // GIVEN
    const app = new App({
        context : {}
    });
    const stack = new LambdaPythonAlphaSamplesStack(app, 'LambdaPythonAlphaSamplesStack', {
        stackName: 'LambdaPythonAlphaSamplesStack',
        description: 'Lambda Python Alpha Samples Stack',
        env: defaultEnv,
      });
      
    // WHEN
    const template = Template.fromStack(stack);
    // THEN
    // test with snapshot
    expect(template).toMatchSnapshot();

});