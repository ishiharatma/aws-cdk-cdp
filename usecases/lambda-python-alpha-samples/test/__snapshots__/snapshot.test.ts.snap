// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Description": "Lambda Python Alpha Samples Stack",
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "LambdaPythonAlphaSamplesDEBUG4A0E212B": {
      "DependsOn": [
        "LambdaPythonAlphaSamplesDEBUGServiceRoleDefaultPolicy6F15290E",
        "LambdaPythonAlphaSamplesDEBUGServiceRole6BC491E8",
        "LambdaPythonAlphaSamplesDEBUGLogGroup2236A40F",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-123456789012-ap-northeast-1",
          "S3Key": "9483d39f3a0675406d784e96f564125d9b5c61a9e7535bd79538f0d116958a2d.zip",
        },
        "Description": "Lambda Python Alpha Samples DEBUG",
        "Environment": {
          "Variables": {
            "MY_ENV_VAR": "my_value",
            "MY_ENV_VAR2": "my_value2",
          },
        },
        "FunctionName": "LambdaPythonAlphaSamplesDEBUG",
        "Handler": "index.lambda_handler",
        "LoggingConfig": {
          "ApplicationLogLevel": "DEBUG",
          "LogFormat": "JSON",
          "LogGroup": {
            "Ref": "LambdaPythonAlphaSamplesDEBUGLogGroup2236A40F",
          },
        },
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "LambdaPythonAlphaSamplesDEBUGServiceRole6BC491E8",
            "Arn",
          ],
        },
        "Runtime": "python3.13",
        "Timeout": 30,
        "TracingConfig": {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaPythonAlphaSamplesDEBUGLogGroup2236A40F": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "LogGroupName": "/aws/lambda/LambdaPythonAlphaSamplesDEBUG",
        "RetentionInDays": 1,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "LambdaPythonAlphaSamplesDEBUGServiceRole6BC491E8": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaPythonAlphaSamplesDEBUGServiceRoleDefaultPolicy6F15290E": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LambdaPythonAlphaSamplesDEBUGServiceRoleDefaultPolicy6F15290E",
        "Roles": [
          {
            "Ref": "LambdaPythonAlphaSamplesDEBUGServiceRole6BC491E8",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "LambdaPythonAlphaSamplesERROR324F8EB6": {
      "DependsOn": [
        "LambdaPythonAlphaSamplesERRORServiceRoleDefaultPolicy97DCCDDD",
        "LambdaPythonAlphaSamplesERRORServiceRoleADDC0E82",
        "LambdaPythonAlphaSamplesERRORLogGroupB084BE42",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-123456789012-ap-northeast-1",
          "S3Key": "9483d39f3a0675406d784e96f564125d9b5c61a9e7535bd79538f0d116958a2d.zip",
        },
        "Description": "Lambda Python Alpha Samples ERROR",
        "Environment": {
          "Variables": {
            "MY_ENV_VAR": "my_value",
            "MY_ENV_VAR2": "my_value2",
          },
        },
        "FunctionName": "LambdaPythonAlphaSamplesERROR",
        "Handler": "index.lambda_handler",
        "LoggingConfig": {
          "ApplicationLogLevel": "ERROR",
          "LogFormat": "JSON",
          "LogGroup": {
            "Ref": "LambdaPythonAlphaSamplesERRORLogGroupB084BE42",
          },
        },
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "LambdaPythonAlphaSamplesERRORServiceRoleADDC0E82",
            "Arn",
          ],
        },
        "Runtime": "python3.13",
        "Timeout": 30,
        "TracingConfig": {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaPythonAlphaSamplesERRORLogGroupB084BE42": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "LogGroupName": "/aws/lambda/LambdaPythonAlphaSamplesERROR",
        "RetentionInDays": 1,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "LambdaPythonAlphaSamplesERRORServiceRoleADDC0E82": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaPythonAlphaSamplesERRORServiceRoleDefaultPolicy97DCCDDD": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LambdaPythonAlphaSamplesERRORServiceRoleDefaultPolicy97DCCDDD",
        "Roles": [
          {
            "Ref": "LambdaPythonAlphaSamplesERRORServiceRoleADDC0E82",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "LambdaPythonAlphaSamplesINFO33C182AD": {
      "DependsOn": [
        "LambdaPythonAlphaSamplesINFOServiceRoleDefaultPolicyEBA83756",
        "LambdaPythonAlphaSamplesINFOServiceRole58DAE35B",
        "LambdaPythonAlphaSamplesINFOLogGroup06D3E8AA",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-123456789012-ap-northeast-1",
          "S3Key": "9483d39f3a0675406d784e96f564125d9b5c61a9e7535bd79538f0d116958a2d.zip",
        },
        "Description": "Lambda Python Alpha Samples INFO",
        "Environment": {
          "Variables": {
            "MY_ENV_VAR": "my_value",
            "MY_ENV_VAR2": "my_value2",
          },
        },
        "FunctionName": "LambdaPythonAlphaSamplesINFO",
        "Handler": "index.lambda_handler",
        "LoggingConfig": {
          "ApplicationLogLevel": "INFO",
          "LogFormat": "JSON",
          "LogGroup": {
            "Ref": "LambdaPythonAlphaSamplesINFOLogGroup06D3E8AA",
          },
        },
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "LambdaPythonAlphaSamplesINFOServiceRole58DAE35B",
            "Arn",
          ],
        },
        "Runtime": "python3.13",
        "Timeout": 30,
        "TracingConfig": {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaPythonAlphaSamplesINFOLogGroup06D3E8AA": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "LogGroupName": "/aws/lambda/LambdaPythonAlphaSamplesINFO",
        "RetentionInDays": 1,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "LambdaPythonAlphaSamplesINFOServiceRole58DAE35B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaPythonAlphaSamplesINFOServiceRoleDefaultPolicyEBA83756": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LambdaPythonAlphaSamplesINFOServiceRoleDefaultPolicyEBA83756",
        "Roles": [
          {
            "Ref": "LambdaPythonAlphaSamplesINFOServiceRole58DAE35B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "LambdaPythonAlphaSamplesWARN675A151A": {
      "DependsOn": [
        "LambdaPythonAlphaSamplesWARNServiceRoleDefaultPolicy2A5F0E0E",
        "LambdaPythonAlphaSamplesWARNServiceRole159CF76D",
        "LambdaPythonAlphaSamplesWARNLogGroupAF95260A",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-123456789012-ap-northeast-1",
          "S3Key": "9483d39f3a0675406d784e96f564125d9b5c61a9e7535bd79538f0d116958a2d.zip",
        },
        "Description": "Lambda Python Alpha Samples WARN",
        "Environment": {
          "Variables": {
            "MY_ENV_VAR": "my_value",
            "MY_ENV_VAR2": "my_value2",
          },
        },
        "FunctionName": "LambdaPythonAlphaSamplesWARN",
        "Handler": "index.lambda_handler",
        "LoggingConfig": {
          "ApplicationLogLevel": "WARN",
          "LogFormat": "JSON",
          "LogGroup": {
            "Ref": "LambdaPythonAlphaSamplesWARNLogGroupAF95260A",
          },
        },
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "LambdaPythonAlphaSamplesWARNServiceRole159CF76D",
            "Arn",
          ],
        },
        "Runtime": "python3.13",
        "Timeout": 30,
        "TracingConfig": {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaPythonAlphaSamplesWARNLogGroupAF95260A": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "LogGroupName": "/aws/lambda/LambdaPythonAlphaSamplesWARN",
        "RetentionInDays": 1,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "LambdaPythonAlphaSamplesWARNServiceRole159CF76D": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaPythonAlphaSamplesWARNServiceRoleDefaultPolicy2A5F0E0E": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LambdaPythonAlphaSamplesWARNServiceRoleDefaultPolicy2A5F0E0E",
        "Roles": [
          {
            "Ref": "LambdaPythonAlphaSamplesWARNServiceRole159CF76D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
